lib/model/rwfolder.go

```diff
@@ -87,6 +89,8 @@ type rwFolder struct {
 	copiers      int
 	pullers      int
 	shortID      uint64
+	newVersion   protocol.Vector
```

Comment:
Ok so... I tried to rework the code but I'm lost.  I created an object in rwFolder struct of type sharedPullerState instead of the ones that were there before, but I have no idea what to do with it to populate it with data.  And of course if I try to read the objects in debug log I see nothing in them (probably because they aren't populated).  I don't know what function I could use within rwFolder to populate the sharedPullerState object I created.  Or if that's even the right way to get this done.